src/Microsoft.Performance.SDK.Runtime/CustomDataSourceReferenceExtensions.cs

```diff
@@ -11,45 +14,101 @@ namespace Microsoft.Performance.SDK.Runtime
     public static class CustomDataSourceReferenceExtensions
     {
         /// <summary>
-        ///     Extension method to get the canonical file extension for a <see cref="CustomDataSourceReference"/>.
+        ///     Extension method to get the file extensions for a <see cref="CustomDataSourceReference"/>.
         /// </summary>
         /// <returns>
-        ///     <inheritdoc cref="FileExtensionUtils.CanonicalizeExtension(string)"/>
+        ///     All file extensions supported by the referenced Custom Data Source.
         /// </returns>
-        public static string TryGetCanonicalFileExtension(this CustomDataSourceReference self)
+        public static IEnumerable<string> TryGetFileExtensions(this CustomDataSourceReference self)
         {
-            var fileDataSource = self.DataSource as FileDataSourceAttribute;
-            var fileDataSourceExtension = fileDataSource?.FileExtension;
+            return self.DataSources
+                .OfType<FileDataSourceAttribute>()
+                .Select(x => x.FileExtension);
+        }
 
-            return FileExtensionUtils.CanonicalizeExtension(fileDataSourceExtension);
+        /// <summary>
+        ///     Extension method to get the canonical file extensions for a <see cref="CustomDataSourceReference"/>.
+        /// </summary>
+        /// <returns>
+        ///     The canonical forms of the file extensions supported by the referenced
+        ///     Custom Data Source.
+        /// </returns>
+        public static IEnumerable<string> TryGetCanonicalFileExtensions(this CustomDataSourceReference self)
+        {
+            return self.TryGetFileExtensions()
+                .Select(FileExtensionUtils.CanonicalizeExtension);
         }
 
         /// <summary>
         ///     Extension method to get the description for a <see cref="CustomDataSourceReference"/>.
         /// </summary>
+        /// <param name="extension">
+        ///     The extension of the file whose description is to be retreived. Specify
+        ///     <c>null</c> or whitespace to retrieve the description of extensionless files.
+        /// </param>
         /// <returns>
-        ///     <inheritdoc cref="CustomDataSourceExtensions.TryGetFileDescription(ICustomDataSource)"/>
+        ///     The description of the file specified by the given exteions; <c>null</c> if not found.
         /// </returns>
-        public static string TryGetFileDescription(this CustomDataSourceReference self)
+        public static string TryGetFileDescription(this CustomDataSourceReference self, string extension)
```

Comment:
Since we have `TryGetDirectoryDescription`, would it make sense to have a `TryGetExtensionlessFileDescription` too? Just so there's a bijection between the three built-in data sources and these methods

Reply:
Yes, I'll add it.